dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/DynamodbEnhancedClientRecorder.java

```diff
@@ -1,21 +1,40 @@
 package io.quarkus.amazon.dynamodb.enhanced.runtime;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
 import io.quarkus.arc.runtime.BeanContainer;
 import io.quarkus.arc.runtime.BeanContainerListener;
 import io.quarkus.runtime.annotations.Recorder;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClientExtension;
+import software.amazon.awssdk.enhanced.dynamodb.internal.client.ExtensionResolver;
 import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 
 @Recorder
 public class DynamodbEnhancedClientRecorder {
+    private static final Log LOG = LogFactory.getLog(DynamodbEnhancedClientRecorder.class);
+
+    DynamoDbEnhancedBuildTimeConfig buildTimeConfig;
+
+    public DynamodbEnhancedClientRecorder(DynamoDbEnhancedBuildTimeConfig buildTimeConfig) {
+        this.buildTimeConfig = buildTimeConfig;
+    }
 
     public BeanContainerListener setDynamoDbClient() {
         BeanContainerListener beanContainerListener = new BeanContainerListener() {
 
             @Override
             public void created(BeanContainer container) {
                 DynamodbEnhancedClientProducer producer = container.beanInstance(DynamodbEnhancedClientProducer.class);
-                producer.setDynamoDbClient(container.beanInstance(DynamoDbClient.class));
+                producer.setDynamoDbClient(container.beanInstance(DynamoDbClient.class), createExtensionList());
```

Comment:
I would createExtensionList sooner to fail faster in case of misconfiguration in the recorder method and pass the value as closure to the listener. Or if we want to share the same instance for both sync and async, we can have the createExtensionList return a `RuntimeValue<DynamoDbEnhancedClientExtension>` and have it inject from the processor.

In recorder, something like
```
return  new RuntimeValue<>(ExtensionResolver.resolveExtensions(extensions));
````

and in the processor

```
var extensions = recorder.createExtensionList();

containerListenerProducer.produce(new BeanContainerListenerBuildItem(recorder.setDynamoDbClient(extensions )));
```

Reply:
Great! Thanks for this advice! I took your tips and implement it.